staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go

```diff
@@ -100,13 +100,13 @@ func testPropagateStore(ctx context.Context, t *testing.T, store storage.Interfa
 	return key, setOutput
 }
 
-func ExpectNoDiff(t *testing.T, msg string, expected, got interface{}) {
+func expectObjectsDeepEqual(t *testing.T, msg string, expected, actual interface{}) {
 	t.Helper()
-	if !reflect.DeepEqual(expected, got) {
-		if diff := cmp.Diff(expected, got); diff != "" {
-			t.Errorf("%s: %s", msg, diff)
+	if !reflect.DeepEqual(expected, actual) {
+		if diff := cmp.Diff(expected, actual); diff != "" {
+			t.Fatalf("%s: %s", msg, diff)
```

Comment:
please revert - I actually prefer t.Errorf, as in case of multiple events we may get more detailed results.

Reply:
the issue is that the output is not readable since this function is used in unit tests it is easier to fail early just like `require.NoError` function does.